world.py

```diff
@@ -71,6 +73,9 @@ def step(self):
                         dead.add(candidate)
                         creature.energy += candidate.energy
                         continue
+                    else:
+                        # I would say code can never get here
+                        assert False, f"candidate.energy {candidate.energy} should not be larger than creature.energy {creature.energy}"
```

Comment:
Suggest to replace 67 with an assert `isinstance` which would allow us to not have the `if` clause